extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -10,46 +10,79 @@ class NodeImage extends Node {
+		$file = $this->getFilefromTitle( $title );
+		if ( $title instanceof \Title ) {
+			$this->getExternalParser()->addImage( $title->getDBkey() );
+		}
+		$ref = null;
+		wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
+		$item = [
+			'url' => $this->resolveImageUrl( $file ),
+			'name' => ( $title ) ? $title->getText() : '',
+			'key' => ( $title ) ? $title->getDBKey() : '',
+			'alt' => $alt,
+			'caption' => $caption,
+			'ref' => $ref
+		];
+		if ( $this->isVideo( $file ) ) {
+			$item = $this->videoDataDecorator( $item, $file );
+		}
+		return $item;
+	}
+
 	public function getData() {
 		if ( !isset( $this->data ) ) {
 			$imageData = $this->getRawValueWithDefault( $this->xmlNode );
-
-			if( is_string($imageData) && PortableInfoboxDataBag::getInstance()->getGallery($imageData)) {
-				$imageData = PortableInfoboxDataBag::getInstance()->getGallery($imageData);
-			}
-
-			$title = $this->getImageAsTitleObject( $imageData );
-			$file = $this->getFilefromTitle( $title );
-			if ( $title instanceof \Title ) {
-				$this->getExternalParser()->addImage( $title->getDBkey() );
-			}
-			$ref = null;
-			$alt = $this->getValueWithDefault( $this->xmlNode->{self::ALT_TAG_NAME} );
-			$caption = $this->getValueWithDefault( $this->xmlNode->{self::CAPTION_TAG_NAME} );
-
-			wfRunHooks( 'PortableInfoboxNodeImage::getData', [ $title, &$ref, $caption ] );
-
-			$this->data = [
-				'url' => $this->resolveImageUrl( $file ),
-				'name' => ( $title ) ? $title->getText() : '',
-				'key' => ( $title ) ? $title->getDBKey() : '',
-				'alt' => $alt,
-				'caption' => $caption,
-				'ref' => $ref
-			];
-
-			if ( $this->isVideo( $file ) ) {
-				$this->data = $this->videoDataDecorator( $this->data, $file );
+			$imageDataLower = strtolower($imageData);
+
+			if ( strpos( $imageDataLower, '-tabber-' ) !== false || strpos( $imageDataLower, '-gallery-' ) !== false ) {
+				$parsed = $this->getExternalParser()->parseRecursive( $imageData );
+
+				// TODO: Move to separated function + error handling
+				$galleryImages = array();
+				if ( preg_match('#\sdata-model="([^"]+)"#', $parsed, $galleryOut) ) {
+					$galleryImages = array_map(function($gallery) {
+						return array('key' => $gallery['dbKey'], 'caption' => $gallery['caption']);
+					}, json_decode(htmlspecialchars_decode($galleryOut[1]), true));
+				}
+
+				// TODO: Move to separated function + error handling
+				$tabberImages = array();
+				if ( preg_match_all('/class="tabbertab" title="([^"]+)".*?\sdata-image-key="([^"]+)"/is', $parsed, $tabberOut) ) {
+					for( $i = 0; $i < count( $tabberOut[0] ); $i++ ) {
+						$tabberImages[] = array('key' => $tabberOut[2][$i], 'caption' => $tabberOut[1][$i]);
+					}
+				}
+				$this->data = [];
+				$images = $galleryImages + $tabberImages;
+				for( $i = 0; $i < count( $images ); $i++ ) {
+					$this->data[] = $this->getImage(
+						$images[$i]['key'],
+						$images[$i]['caption'],
```

Comment:
I don't think is even needed, AFAIK we do not support alts in galleries nor tabbers, do we? So we can care about caption only